durabletask/worker.py

```diff
@@ -602,17 +690,37 @@ def process_event(self, ctx: _RuntimeOrchestrationContext, event: pb.HistoryEven
                         self._logger.warning(
                             f"{ctx.instance_id}: Ignoring unexpected subOrchestrationInstanceFailed event with ID = {task_id}.")
                     return
-                sub_orch_task.fail(
-                    f"Sub-orchestration task #{task_id} failed: {failedEvent.failureDetails.errorMessage}",
-                    failedEvent.failureDetails)
-                ctx.resume()
+                if isinstance(sub_orch_task, task.RetryableTask):
```

Comment:
Looks like there's a lot of duplicate logic here. I wonder if there's a way we can share more of the logic so that fixes to retry behavior automatically apply to both activities and sub-orchestrations.